.github/workflows/ci.yml

```diff
@@ -78,6 +78,18 @@ jobs:
     steps:
     - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
 
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+    - name: Clean build artifacts
+      run: |
+        cargo clean
```

Comment:
Can you add a trailing newline here (like we do between other steps)?

Reply:
Sure, I will add it.

Reply:
(Commenting here to turn this into a review comment)

Reply:
Same comment as below: Is running `cargo clean` necessary here?